api/v1alpha2/bundledeployment_types.go

```diff
@@ -69,8 +69,42 @@ type BundleDeploymentSpec struct {
 	//
 	// config is provisioner specific configurations
 	Config runtime.RawExtension `json:"config,omitempty"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be enabled or disabled when attempting to install the BundleDeployment.
+	Preflight *PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
+type PreflightConfig struct {
+	//+kubebuilder:Optional
+	CRDUpgradeSafety *CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
+}
+
+// CRDUpgradeSafetyPreflightConfig is a custom struct that holds the necessary configuration values to
+// enable or disable the CRD Upgrade Safety validations. Currently, this holds Mode
+// that sets the CRD Upgrade Safety validations to either Enabled/Disabled.
+// By default, this is set to "Enabled".
+type CRDUpgradeSafetyPreflightConfig struct {
+	//+kubebuilder:validation:Enum:=Enabled;Disabled
+	//+kubebuilder:default:=Enabled
+	//+kubebuilder:Optional
+	Mode *CRDUpgradeSafetyMode `json:"mode,omitempty"`
```

Comment:
`Mode` should have it's own description comment. This is where we can talk about `Enabled` being the default value.